targets/csharp/source/source/PlayFabHttp/PlayFabHttp.cs

```diff
@@ -41,28 +41,26 @@ public class PlayFabJsonSuccess<TResult> where TResult : PlayFabResultCommon
 
     public static class PlayFabHttp
     {
-        private static readonly IPlayFabHttp _http;
-
-        static PlayFabHttp()
+        public static async Task<object> DoPost(string urlPath, PlayFabRequestCommon request, string authType, string authKey, Dictionary<string, string> extraHeaders)
         {
-            var httpInterfaceType = typeof(IPlayFabHttp);
-            var types = typeof(PlayFabHttp).GetAssembly().GetTypes();
-            foreach (var eachType in types)
+            if (PlayFabSettings.TitleId == null)
+                throw new Exception("You must set your titleId before making an api call");
+            var transport = (ITransportPlugin)PluginManager.GetPlugin(PluginContract.PlayFab_Transport);
+
+            var headers = new Dictionary<string, string>();
+            if (authType != null)
```

Comment:
Just talked offline about this. We aren't throwing in that case, because some methods like Login*** may not require those headers

---
In reply to: [201520023](https://github.com/PlayFab/SDKGenerator/pull/171#discussion_r201520023) [](ancestors = 201520023)